include/envoy/network/connection.h

```diff
@@ -120,14 +120,15 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * @return The address of the remote client.
    */
   virtual const Address::Instance& remoteAddress() PURE;
-
+  virtual Address::InstanceConstSharedPtr remoteAddressConstSharedPtr() PURE;
   /**
    * @return the local address of the connection. For client connections, this is the origin
    * address. For server connections, this is the local destination address. For server connections
    * it can be different from the proxy address if the downstream connection has been redirected or
    * the proxy is operating in transparent mode.
    */
   virtual const Address::Instance& localAddress() PURE;
+  virtual Address::InstanceConstSharedPtr localAddressConstSharedPtr() PURE;
```

Comment:
Sorry I'm a little lost. Yeah the addition of these new functions is pretty strange IMO. Why is this needed? I would need to look through all the code again, but the IPList/CidrRange construction is not performance critical. Only isContains() etc. is perf critical. Why are pointers needed at all? Can't we just pass the address reference and do the check?